std/utf8.zig

```diff
@@ -0,0 +1,144 @@
+const unicode = @import("unicode.zig");
+const Rune = unicode.Rune;
+
+const Errors = error {
+    InvalidUtf8,
+};
+
+// Returns true if the byte slice contains valid utf8.
+pub fn validate(s: []const u8) bool {
+    var i: usize = 0;
+    while (i < s.len) {
+        if (unicode.utf8ByteSequenceLength(s[i])) |cp_len| {
+            i += cp_len;
+        } else |err| {
+            return false;
+        }
+    }
+    return true;
```

Comment:
`if (i > s.len) return false;`

You need this check in case a multi-byte sequence would overflow the end of the buffer.